code/script.js

```diff
@@ -1,325 +1,824 @@
+
+  // check a name of secret and guessed person's name is the same
+  if (personToCheck === secret.name) {
+    handleLocalStorage();
+
+    winOrLoseText.textContent = `✨🎉Conglaturation!! 🎉✨`;
+    createSound("./images/audio/win-sound.mp3");
+  } else {
+    winOrLoseText.textContent = `💫 Oh no!! A wrong person. 
+                                        Game over 😱 `;
+    createSound("./images/audio/fail-sound.mp3");
+  }
+};
+
+// Handle Local Storage
+
+const handleLocalStorage = () => {
+  const dataStorage = {
+    playerName: player,
+    howManyGuesses: finalCounts,
+    time: finalTimerValue,
+  };
+  // Pushing a newly created obj with game data into the array that stores data from localstorage
+  arrayForLocalStorage.push(dataStorage);
+
+  // then store again in localstorage
+  window.localStorage.setItem("items", JSON.stringify(arrayForLocalStorage));
+};
+
+/*********************************************************************/
+// opening the borad page
+const startGameBtn = document.getElementById("game-start-btn");
+const initialPage = document.getElementById("initial-page");
+
+player = prompt("Who is going to play, today?");
+
+// if player did not type its name, alert and ask again
+while (!player) {
+  alert("Please type your name!");
+  player = prompt("Who is going to play,today");
 }
 
-// Invokes the start function when website is loaded
-start()
+welcomeMessage.innerText = `Welcome to Guess Who, ${player}!! `;
+
+startGameBtn.onclick = function () {
+  initialPage.style.display = "none";
+
+  NameContainer.innerText = `Player: ${player}`;
+  // Invokes the start function when startgameBtn is clicked
+  start();
+};
+
+/*********************************************************************/
+// Sound effect function
+
+function createSound(soundSrc) {
+  const audio = document.createElement("audio");
+  winOrLosePage.appendChild(audio);
+  audio.setAttribute("src", soundSrc);
+  audio.volume = 0.1;
+  audio.play();
+}
+
+/*********************************************************************/
+// timer function
+
+const startTimer = () => {
+  console.log("start Timer");
+  clearInterval(timerInterval);
+
+  let second = 0,
+    minute = 0,
+    hour = 0;
+
+  timerInterval = setInterval(function () {
+    timer.innerHTML =
+      (hour ? hour + ":" : "") +
+      (minute < 10 ? "0" + minute : minute) +
+      ":" +
+      (second < 10 ? "0" + second : second);
+
+    second++;
+
+    if (second == 60) {
+      minute++;
+      second = 0;
+    }
+    if (minute == 60) {
+      hour++;
+      minute = 0;
+    }
+  }, 1000);
+};
+/*********************************************************************/
+// For Modal Window
+
+// Here I am getting data from localStorage
+const getDataFromLocalStorage = () => {
+  if (localStorage.length === 1) {
+    modalDisplayArr = JSON.parse(localStorage.getItem("items"));
+  }
+};
+
+// checking the best scored game
+const checkBextGame = (arr) => {
```

Comment:
Great feature to have a score board! It could incourage the player to play more times to beat the high score.